software/controller/lib/core/alarm.cpp

```diff
@@ -0,0 +1,102 @@
+#include <stdint.h>
+#include "vars.h"
+#include "debug.h"
+#include "hal.h"
+#include "alarm.h"
+#include "comms.h"
+#include "commfail_alarm.h"
+
+// MAX_CONTROLLER_ALARMS 
+#define MAX_CONTROLLER_ALARMS 2
+
+// structure definition for MAX_CONTROLLER_ALARMS 
+alarm_cb_t alarms[MAX_CONTROLLER_ALARMS];
+
+// this value is reset to 0 after MAX_CONTROLLER_ALARMS 
+static uint16_t alarm_check_counter;
+
+// There should be a packet from GUI for every 30ms. Assuming something went
+// wrong, we wait for 50ms before raising a communication failure alarm
+static constexpr Duration COMMUNICATION_TIMEOUT = milliseconds(10000);
```

Comment:
Is the same 10000 that is later used in `alarm_handler`?
Also, should follow Google style for naming of constants:
https://google.github.io/styleguide/cppguide.html#Constant_Names

Also, variables with units, should have units in the name, in this case milliseconds.

Reply:
This is communication timeout duration. used similar to the usage as per in comms.cpp

Reply:
Not sure which aspect of the question you are addressing here.
A) Naming should follow the style guide. If other parts of the code are not yet following it perfectly, it is not an indication of an exception. Should still follow style. Fu
B) Did not add indication of units to the name, milliseconds, if that is what they are.
C) And how about is this the same 10000 as used later?